File: gigachadsys-model/src/main/java/pe/edu/pucp/gigachadsys/model/Socio.java
package pe.edu.pucp.gigachadsys.model;

public class Socio extends  Usuario{
    //Atributos
    private boolean estadoMembresia;

    //Constructores
    public Socio() {
        super();
    }

    public Socio(int idUsuario, String nombres, String apellidoMaterno, String apellidoPaterno,
                 int edad, int dni, String email, int telefono, String contrasenia, String rol,
                 boolean estadoMembresia) {

        super(idUsuario, nombres, apellidoMaterno, apellidoPaterno, edad, dni, email, telefono, contrasenia, rol);
        this.estadoMembresia = estadoMembresia;
    }

    //Setters y Getters
    public boolean isEstadoMembresia() {
        return estadoMembresia;
    }

    public void setEstadoMembresia(boolean estadoMembresia) {
        this.estadoMembresia = estadoMembresia;
    }
	

    //Metodos
    @Override
    public void mostrarDatos() {
		
    }
	
	public boolean verificarEstadoActivo(){
		return true;
	}
	
	
}

File: C#/GigaChadSys/GigaChadSys-Model/Socio.cs
namespace GigaChadSysModel;

public class Socio : Usuario
{
    public bool EstadoMembresia { get; set; }

    public List<Reserva> Reservas { get; set; } = new();
    public List<Suscripcion> Suscripciones { get; set; } = new();

    public Socio() { }

    public Socio(int idUsuario, string nombreCompleto, string dni, string email,
                 string telefono, string contrasenia, bool estadoMembresia,
                 bool activo)
        : base(idUsuario, nombreCompleto, dni, email, telefono, contrasenia, activo)
    {
        EstadoMembresia = estadoMembresia;
    }
}